### PR TITLE
Add mobile element insertion identification workflow

### LINKED ISSUE #77: Add mobile element insertion analysis
**New feature**
 Add option to run mobile element insertion analysis.

**Additional info**
*Background (MEIs):* Mobile elements can be transcribed and re-integrated into - predominantly transcriptionally active - DNA regions. Well described for somatic alteration they also play a critical role in germline variation.

*Ideas:* Use soft-clipped bases information in alignment file to cluster for mobile element sequences found (Tool: scramble [1], For background see also [2]).

[1] https://github.com/GeneDx/scramble 
[2] Torene, R.I., Galens, K., Liu, S. et al. Mobile element insertion detection in 89,874 clinical exomes. Genet Med 22, 974–978 (2020) [[link](https://www.nature.com/articles/s41436-020-0749-x)]

### PR DESCRIPTION
**Pull request by eudesbarbosa**
closes #77

**Changes**
* Added mobile element insertion identification workflow, tool: `scramble`.

**Out of scope**
The results are per sample, but there is no easy way to merge the genotypes and by extensions the pedigrees. For future iterations a new step to merge MEIs based on overlap would have to be developed.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -48,6 +48,7 @@ Project Info
     step_generic
     step/hla_typing
     step/igv_session_generation
+    step/mobile_element_insertion
```
Comment (holtgrewe) [inline docs/index.rst:51]: Should probably be called targeted_seq_mei_calling as we have wgs_mei_calling already.

File: snappy_pipeline/workflows/mobile_element_insertion/__init__.py [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+        """Get parameters.
+
+        :param _wildcards: Snakemake rule wildcards (unused).
+        :type _wildcards: snakemake.io.Wildcards
+
+        :return: Returns parameters required to run analysis part of scramble.
+        """
+        # Define required paths
+        base_path = self.config["scramble_install_dir"]
+        rscript_path = os.path.join(base_path, "SCRAMble.R")
```
Comment (holtgrewe) [inline snappy_pipeline/workflows/mobile_element_insertion/__init__.py:263]: Can we switch to using scrambl package from bioconda?

https://bioconda.github.io/recipes/scramble/README.html?highlight=scramble

File: tests/snappy_pipeline/workflows/test_workflows_mobile_element_insertion.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    # Define expected
+    pattern_out = "work/{mapper}.scramble.{library_name}/out/{mapper}.scramble.{library_name}"
+    expected = {"txt": pattern_out + "_MEIs.txt", "txt_md5": pattern_out + "_MEIs.txt.md5"}
+    # Get actual
+    actual = mei_workflow.get_output_files("scramble", "analysis")
+    assert actual == expected
+
+
+def test_scramble_analysis_step_part_get_log_file(mei_workflow):
+    """Tests ScrambleStepPart::_get_log_files_analysis()"""
```
Comment (holtgrewe) [inline tests/snappy_pipeline/workflows/test_workflows_mobile_element_insertion.py:153]: Python syntax rather is `Class.method`

File: snappy_pipeline/workflows/mobile_element_insertion/__init__.py [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+===========
+Step Output
+===========
+
+For all samples, MEI identification will be performed on the primary DNA NGS libraries separately
+for each configured read mapper and mobile element identification tool. The name of the primary DNA
+NGS library will be used as an identification token in the output file.
+
+For each read mapper, MEI tool, and sample, the following files will be generated:
+
```
Comment (holtgrewe) [inline snappy_pipeline/workflows/mobile_element_insertion/__init__.py:30]: Can we convert the output to VCF?

**Review by holtgrewe — COMMENTED** [review]
Generally, good job. I have the following major points:

- we should use bioconda for tools such as scrambl
- can we generate VCF with scrambl output? VCF 4.x have instructions how to describe the MEI
- the output is per sample, right? can we somehow have per-sample genotyping so one can check which parent also carries the mei or how many individuals carry a MEI?

File: snappy_pipeline/workflows/mobile_element_insertion/__init__.py (hunk 4 above)
Comment (eudesbarbosa) [reply snappy_pipeline/workflows/mobile_element_insertion/__init__.py]: I will put this change on hold. `scramble` has some internal issue with a non informative error message when I ask to output VCF...
**command:**
```
scramble.sh --ref /fast/projects/cubit/20.05/static_data/reference/GRCh37/hs37d5/hs37d5.fa \
			--out-name SAMPLE_PREFIX \
			--cluster-file /path/to/SAMPLE_cluster.txt \
			--nCluster 5 \
			--mei-score 50 \
			--indel-score 80 \
			--poly-a-frac 0.75 \
			--eval-meis
```
**error:**
```
Done analyzing l1 
Done analyzing sva 
Done analyzing alu 
Done analyzing l1 
Done analyzing sva 
Done analyzing alu 
Sample had 137 MEI(s)
Done analyzing MEIs
Loading required package: GenomicRanges
Error: subscript contains invalid names
Execution halted
```

Comment (eudesbarbosa) [reply snappy_pipeline/workflows/mobile_element_insertion/__init__.py]: Done.

File: snappy_pipeline/workflows/targeted_seq_mei_calling/Snakefile [hunk 5]
```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+"""CUBI Pipeline mobile_element_insertion step Snakefile"""
+
+import os
+
+from snappy_pipeline import expand_ref
+from snappy_pipeline.workflows.targeted_seq_mei_calling import MEIWorkflow
```
Comment (holtgrewe) [inline snappy_pipeline/workflows/targeted_seq_mei_calling/Snakefile:7]: Should this be `MeiWorkflow`?

**Review by holtgrewe — APPROVED** [review]
LGTM

File: snappy_pipeline/workflows/targeted_seq_mei_calling/Snakefile (hunk 5 above)
Comment (eudesbarbosa) [reply snappy_pipeline/workflows/targeted_seq_mei_calling/Snakefile]: Modified.

**Pull request merged**